moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/include/moveit/hybrid_planning_manager/hybrid_planning_manager.h

```diff
@@ -107,7 +107,7 @@ class HybridPlanningManager : public rclcpp::Node
   bool initialized_;
 
   // Flag that indicates hybrid planning has been canceled
-  bool stop_hybrid_planning_;
+  std::atomic<bool> stop_hybrid_planning_;
```

Comment:
sadness... raw synchronization primitive and using mutable state

Reply:
You might find this interesting: https://en.wikipedia.org/wiki/Amdahl%27s_law

There are two reasons to remove shared memory between threads.  

The first is that it is a synchronization point (a point in the software that must be single-threaded) and that the more of those you add the worse our software can scale across multiple processors.  See above link.

The second reason is that they are mutable state.  See the C++ seasoning talk for more reasoning on that.

Reply:
My understanding is atomics are lock-free since the read or write happens in a single step. So the first bullet point doesn't apply. I'm not sure about the second one.

Reply:
To make it clear from our discussion today I want to find a better way to communicate between threads than raw synchronization primitives around memory.  That however is a much larger discussion and requires more research than this PR.  Please just make sure you are using some mechanism (atomic, mutex, etc) to synchronize access to any variables you are sharing between two concurrent sections.